core/src/main/java/org/mule/transport/AbstractConnector.java

```diff
@@ -290,6 +292,8 @@
      */
     private boolean validateConnections = true;
 
+    private Lock lock = new ReentrantLock();
```

Comment:
This look is used only on connection/reconnection. Change the name to make that more explicit